src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/RewriteMapParser.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Microsoft.AspNetCore.Rewrite.Internal.IISUrlRewrite
+{
+    public static class RewriteMapParser
+    {
+        public static IISRewriteMapCollection Parse(XElement xmlRoot)
+        {
+            if (xmlRoot == null)
+            {
+                throw new ArgumentException(nameof(xmlRoot));
+            }
+
+            var mapsElement = xmlRoot.Descendants(RewriteTags.RewriteMaps).SingleOrDefault();
+            if (mapsElement == null)
+            {
+                return null;
```

Comment:
i could move the test for the element outside of the parser and never end up in that scenario but this seems to be the preferred style throughout the project. i definitely don't think it should throw, though. is the structure what's objectionable?